### PR TITLE
Fix/permissions for work package index

### PR DESCRIPTION
**Pull request by manwithtwowatches**
https://www.openproject.org/work_packages/10435

Need to write some tests.

### CODE REVIEW

**Comment by ulferts** [comment]
The "+ Work package"-button is not disabled if the user lacks the permission to create work packages. The user can thus cklick on the link to create e.g. a new task and is redirected to the login page. We should prevent the user from going there in the first place.

The same is true for the "+ Work package"-button button on the global work packages index page, where creating a work package is never possible. There, an empty drop down is displayed. But it should be fixed along with the other problem because the "create" link will be missing in the server response there, too.

File: app/assets/javascripts/angular/work_packages/controllers/work-packages-list-controller.js [hunk 1]
```diff
@@ -139,6 +140,9 @@ angular.module('openproject.workPackages.controllers')
 
     // back url
     $scope.updateBackUrl();
+
+    // Authorisation
+    $scope.disableNewWorkPackage = AuthorisationService.cannot({ links: meta._links }, "create");
```
Comment (manwithtwowatches) [inline app/assets/javascripts/angular/work_packages/controllers/work-packages-list-controller.js:146]: @ulferts this is slightly misleading as the meta data from the work packages index isn't a model like AuthorisationService.cannot is expecting, but i think this approximates how the data will be returned form the hateos api.

Comment (ulferts) [inline app/assets/javascripts/angular/work_packages/controllers/work-packages-list-controller.js:146]: fine with that

**Comment by ulferts** [comment]
Just one tiny proposal. Can't we disable the button by default and enable if the user has the permission?

Based on how long the query takes to proceed, the user might be tempted to click on the active button which is only disabled after the response is received and interpreted.

**Comment by manwithtwowatches** [comment]
Yeah sure. Also there is an error being thrown right now that I hadn't noticed which I'm fixing.

**Pull request merged**